core/src/test/scala/unit/kafka/admin/ReassignPartitionsClusterTest.scala

```diff
@@ -103,11 +108,60 @@ class ReassignPartitionsClusterTest extends ZooKeeperTestHarness with Logging {
     assertEquals(actual.values.flatten.toSeq.distinct.sorted, Seq(100, 101))
   }
 
+  @Test
+  def shouldChangeThrottleOnRerun() {
+    //Given partitions on 3 of 3 brokers
+    val brokers = Array(100, 101, 102)
+    startBrokers(brokers)
+    createTopic(zkUtils, topicName, Map(
+      0 -> Seq(100, 101)
+    ), servers = servers)
+
+    //Given throttle set so replication will take at least 20 sec (we won't wait this long)
+    val initialThrottle: Long = 1000 * 1000
+    addMessages(num = 200, size = 100 * 1000, topicName, servers)
+
+    //Start rebalance (use a separate thread as it'll be slow)
+    val newAssignment = ReassignPartitionsCommand.generateAssignment(zkUtils, Array(101, 102), json(topicName), true)._1
+
+    inThread() {
+      ReassignPartitionsCommand.executeAssignment(zkUtils, ZkUtils.formatAsReassignmentJson(newAssignment), initialThrottle)
+      println("Runnable 1 completed")
+    }
+
+    //Check throttle config
+    checkThrottleConfigAddedToZK(initialThrottle, servers, topicName)
+
+    //Now re-run the same assignment with a larger throttle, which should only act to increase the throttle and make progress (again use a thread so we can check ZK whilst it runs)
+    val newThrottle = initialThrottle * 1000
+
+    inThread() {
+      ReassignPartitionsCommand.executeAssignment(zkUtils, ZkUtils.formatAsReassignmentJson(newAssignment), newThrottle)
+    }
+
+    //Check throttle was changed
+    checkThrottleConfigAddedToZK(newThrottle, servers, topicName)
+
+    //Await completion
+    waitForReasignmentToComplete()
+
+    //Check removed
+    checkThrottleConfigRemovedFromZK(topicName, servers)
+
+    //Check move occurred
+    val actual = zkUtils.getPartitionAssignmentForTopics(Seq(topicName))(topicName)
+    assertEquals(actual.values.flatten.toSeq.distinct.sorted, Seq(101, 102))
+  }
+
   def waitForReasignmentToComplete() {
     waitUntilTrue(() => !zkUtils.pathExists(ReassignPartitionsPath), s"Znode ${ZkUtils.ReassignPartitionsPath} wasn't deleted")
   }
 
   def json(topic: String): String = {
     s"""{"topics": [{"topic": "$topic"}],"version":1}"""
   }
+
+  def inThread()(fun: => Unit) = {
+    Utils.newThread("ReplicationTestHelperThread", CoreUtils.runnable(fun), false).start
```

Comment:
We are leaking the thread right? If you just want to execute something in the background, you can do something like:

```scala
import scala.concurrent.{Future, ExecutionContext}
import ExecutionContext.Implicits._
...
scala> Future(codeToRunInTheGlobalThreadPool)
```